src/main/java/sevensmurfs/rehub/repository/PasswordResetRepository.java

```diff
@@ -0,0 +1,11 @@
+package sevensmurfs.rehub.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import sevensmurfs.rehub.model.entity.PasswordReset;
+
+import java.util.Optional;
+
+public interface PasswordResetRepository extends JpaRepository<PasswordReset, Long> {
+
+    Optional<PasswordReset> findByToken(String token);
```

Comment:
Here you can already add createdAtLessThan... and as parameter add LocalDateTime.now().minusHours() put what you think is the best.